api/src/main/java/io/strimzi/api/kafka/model/common/metrics/MetricsConfig.java

```diff
@@ -23,14 +23,15 @@
         property = "type")
 @JsonSubTypes({
     @JsonSubTypes.Type(name = JmxPrometheusExporterMetrics.TYPE_JMX_EXPORTER, value = JmxPrometheusExporterMetrics.class),
+    @JsonSubTypes.Type(name = StrimziReporterMetrics.TYPE_STRIMZI_REPORTER_METRICS, value = StrimziReporterMetrics.class)
 })
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @EqualsAndHashCode
 @ToString
 public abstract class MetricsConfig implements UnknownPropertyPreserving {
     private Map<String, Object> additionalProperties;
 
-    @Description("Metrics type. Only 'jmxPrometheusExporter' supported currently.")
+    @Description("Metrics type. Currently supports 'strimziReporterMetrics' and 'jmxPrometheusExporter'.")
```

Comment:
This should be `strimziMetricsReporter` I guess? Both here as well as in the class names.